### PR TITLE
add notice about scaling in pd node

### PR DESCRIPTION
**Pull request by AstroProfundis**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add a notice about scaling PD nodes, this is the same as in "scale tidb cluster with Ansible" document.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions) version specific for 4.0.3
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/tikv/tikv/issues/4319

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: scale-tidb-using-tiup.md [hunk 1]
```diff
@@ -234,6 +234,10 @@ tiup cluster display <cluster-name>
 >
 > 移除 TiDB、PD 节点和移除 TiKV 节点的步骤类似。
 
+> **注意：**
+>
+> TiKV 中的 PD Client 会缓存 PD 节点列表，但目前不会定期自动更新，只有在 PD leader 发生切换或 TiKV 重启加载最新配置后才会更新；为避免 TiKV 缓存的 PD 节点列表过旧的风险，在扩缩容 PD 完成后，PD 集群中应至少包含两个扩缩容操作前就已经存在的 PD 节点成员，如果不满足该条件需要手动执行 PD transfer leader 操作，更新 TiKV 中的 PD 缓存列表。
```
Comment (lilyjazz) [inline scale-tidb-using-tiup.md:239]: @rleungx  为什么「在扩缩容完成后，至少包含 2 个扩缩容操作前就已经存在的 PD 节点成员」啊～我看这篇文档的示例中，PD 就只有 1 个节点。会不会造成误解？

Comment (rleungx) [reply scale-tidb-using-tiup.md:239]: /cc @AstroProfundis BTW, we need to specify the version of this warning.

Comment (AstroProfundis) [reply scale-tidb-using-tiup.md:239]: What would be the version affected?

Comment (rleungx) [reply scale-tidb-using-tiup.md:239]: Almost all versions before v.4.0.3 and we introduce a new mechanism to reduce the probability after v4.0.3

File: scale-tidb-using-tiup.md [hunk 2]
```diff
@@ -234,6 +234,10 @@ tiup cluster display <cluster-name>
 >
 > 移除 TiDB、PD 节点和移除 TiKV 节点的步骤类似。
 
+> **注意：**
+>
+> TiKV 中的 PD Client 会缓存 PD 节点列表，但在 4.0.3 版本之前不会定期自动更新，只有在 PD leader 发生切换或 TiKV 重启加载最新配置后才会更新；为避免 TiKV 缓存的 PD 节点列表过旧的风险，在扩缩容 PD 完成后，PD 集群中应至少包含一个扩缩容操作前就已经存在的 PD 节点成员，如果不满足该条件需要手动执行 PD transfer leader 操作以更新 TiKV 中的 PD 缓存列表。对于 4.0.3 及以后的版本，增加了定时自动更新 PD 节点的机制，可以降低此问题出现的概率，但仍应尽量避免在扩容新 PD 后直接一次性缩容所有扩容前就已经存在的 PD 节点。
```
Comment (TomShawn) [inline scale-tidb-using-tiup.md:239]:
```suggestion
> TiKV 中的 PD Client 会缓存 PD 节点的列表。
>
> + 在 v4.0.3 版本之前，TiKV 不会定期自动更新该 PD 节点列表的缓存，只有在 PD leader 发生切换或 TiKV 重启加载最新配置后才会更新。为避免 TiKV 缓存的 PD 节点列表过旧的风险，在扩缩容 PD 完成后，PD 集群中应至少包含一个扩缩容操作前就已经存在的 PD 节点成员。如果不满足该条件，你需要手动执行 PD transfer leader 操作以更新 TiKV 中的 PD 缓存列表。
>
> + 在 v4.0.3 及以后的版本中，增加了定期自动更新 PD 节点的机制，可以降低 TiKV 缓存的 PD 节点列表过旧这一问题出现的概率。但仍应尽量避免在扩容新 PD 后直接一次性缩容所有扩容前就已经存在的 PD 节点。
```

File: scale-tidb-using-tiup.md [hunk 3]
```diff
@@ -234,6 +234,14 @@ tiup cluster display <cluster-name>
 >
 > 移除 TiDB、PD 节点和移除 TiKV 节点的步骤类似。
 
+> **注意：**
+>
+> TiKV 中的 PD Client 会缓存 PD 节点的列表。
+>
+> + 在 v4.0.3 版本之前，TiKV 不会定期自动更新该 PD 节点列表的缓存，只有在 PD leader 发生切换或 TiKV 重启加载最新配置后才会更新。为避免 TiKV 缓存的 PD 节点列表过旧的风险，在扩缩容 PD 完成后，PD 集群中应至少包含一个扩缩容操作前就已经存在的 PD 节点成员。如果不满足该条件，你需要手动执行 PD transfer leader 操作以更新 TiKV 中的 PD 缓存列表。
+>
+> + 在 v4.0.3 及以后的版本中，增加了定期自动更新 PD 节点的机制，可以降低 TiKV 缓存的 PD 节点列表过旧这一问题出现的概率。但仍应尽量避免在扩容新 PD 后直接一次性缩容所有扩容前就已经存在的 PD 节点。
```
Comment (TomShawn) [inline scale-tidb-using-tiup.md:243]:
```suggestion
> TiKV 中的 PD Client 会缓存 PD 节点的列表。当前 TiKV 有定期自动更新 PD 节点的机制，可以降低 TiKV 缓存的 PD 节点列表过旧这一问题出现的概率。但你应尽量避免在扩容新 PD 后直接一次性缩容所有扩容前就已经存在的 PD 节点。
```

File: scale-tidb-using-tiup.md [hunk 4]
```diff
@@ -234,6 +234,10 @@ tiup cluster display <cluster-name>
 >
 > 移除 TiDB、PD 节点和移除 TiKV 节点的步骤类似。
 
+> **注意：**
+>
+> TiKV 中的 PD Client 会缓存 PD 节点的列表。当前 TiKV 有定期自动更新 PD 节点的机制，可以降低 TiKV 缓存的 PD 节点列表过旧这一问题出现的概率。但你应尽量避免在扩容新 PD 后直接一次性缩容所有扩容前就已经存在的 PD 节点。
```
Comment (TomShawn) [inline scale-tidb-using-tiup.md:239]:
```suggestion
> TiKV 中的 PD Client 会缓存 PD 节点的列表。当前版本的 TiKV 有定期自动更新 PD 节点的机制，可以降低 TiKV 缓存的 PD 节点列表过旧这一问题出现的概率。但你应尽量避免在扩容新 PD 后直接一次性缩容所有扩容前就已经存在的 PD 节点。
```

File: scale-tidb-using-tiup.md [hunk 5]
```diff
@@ -234,6 +234,10 @@ tiup cluster display <cluster-name>
 >
 > 移除 TiDB、PD 节点和移除 TiKV 节点的步骤类似。
 
+> **注意：**
+>
+> TiKV 中的 PD Client 会缓存 PD 节点的列表。当前版本的 TiKV 有定期自动更新 PD 节点的机制，可以降低 TiKV 缓存的 PD 节点列表过旧这一问题出现的概率。但你应尽量避免在扩容新 PD 后直接一次性缩容所有扩容前就已经存在的 PD 节点。
```
Comment (rleungx) [inline scale-tidb-using-tiup.md:239]:
```suggestion
> TiKV 中的 PD Client 会缓存 PD 节点的列表。当前版本的 TiKV 有定期自动更新 PD 节点的机制，可以降低 TiKV 缓存的 PD 节点列表过旧这一问题出现的概率。但你应尽量避免在扩容新 PD 后直接一次性缩容所有扩容前就已经存在的 PD 节点，如需要，请确保在下线所有之前存在的 PD 前将 PD 的 leader 切换至新扩容的 PD 节点。
```

**Review by rleungx — APPROVED** [review]

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**